sunpy/io/special/asdf/tags/map/generic_map.py

```diff
@@ -19,7 +18,7 @@ class GenericMapType(SunPyType):
     def from_tree(cls, node, ctx):
         # Use the factory here to get the correct subclass back
         out_map = sunpy.map.Map(np.asarray(node['data']), node['meta'])
-        out_map.shift(*node['shift'])
+        out_map.shift_reference_coord(*node['shift'])
```

Comment:
Does calling this when the shift is 0 do anything? If it does (i.e. leads to modifying the keys with the same value) it would also be good to check if the shift value in the file is 0 so that we don't manipulate the meta without reason.

Reply:
No, there's no special casing for 0 shift. I could special case it here or in the method itself?

I'm also now quite confused as this method is being called as if it's an in-place operation but the function returns a new map with the shift applied. I'm not sure that an asdf roundtrip test with non-zero shift would've ever passed.

Reply:
Kind of. It increments the `_shift` attribute and then returns a new map with that incremented attribute.

https://github.com/sunpy/sunpy/blob/fa2c427e6a54493ac39d9536a431504ccc04d014/sunpy/map/mapbase.py#L1017

Reply:
heh, then yeh it probably never worked :facepalm: